osu.Game.Tests/Visual/Navigation/TestScenePerformFromScreen.cs

```diff
@@ -113,12 +114,12 @@ public void TestPerformBlockedByDialog(bool confirmed)
             AddAssert("did not perform", () => !actionPerformed);
             AddAssert("only one exit attempt", () => blocker.ExitAttempts == 1);
 
-            AddUntilStep("wait for dialog display", () => Game.Dependencies.Get<DialogOverlay>().IsLoaded);
+            AddUntilStep("wait for dialog display", () => ((Drawable)Game.Dependencies.Get<IDialogOverlay>()).IsLoaded);
```

Comment:
Not sure how to feel about these casts. It's test code, so I'm not going to make a fuss about ti, but maybe at least a good idea to extract the casting logic to a helper property or something? Just something simple like

```csharp
private DialogOverlay dialogOverlay => (DialogOverlay)Game.Dependencies.Get<IDialogOverlay>();
```

Either that or caching could be permitted for the `DialogOverlay` type too, but I like that solution way less because ideally all consumers should only be using the interface (or else N months down the line it turns out that some part of the game can't accept a mock because it was relying on that).

Interested in other views on this in general.

Reply:
Caching as `DialogOverlay` gets messy and I'd rather avoid it personally. I've refactored the tests to reduce the number of casts.